src/hooks/useAccounts.tsx

```diff
@@ -56,170 +50,42 @@ const useAccountsContext = () => {
-    if (evmDerivedAddresses.version !== LOCAL_STORAGE_VERSIONS[LocalStorageKey.EvmDerivedAddresses])
-      saveEvmDerivedAddresses({
-        version: LOCAL_STORAGE_VERSIONS[LocalStorageKey.EvmDerivedAddresses],
-      });
-  }, []);
-
-  const saveEvmDerivedAccount = useCallback(
-    ({
-      evmAddressInner,
-      dydxAddress,
-    }: {
-      evmAddressInner: EvmAddress;
-      dydxAddress?: DydxAddress;
-    }) => {
-      saveEvmDerivedAddresses({
-        ...evmDerivedAddresses,
-        version: LOCAL_STORAGE_VERSIONS[LocalStorageKey.EvmDerivedAddresses],
-        [evmAddressInner]: {
-          ...(evmDerivedAddresses as any)[evmAddressInner],
-          dydxAddress,
-        },
-      });
-    },
-    [evmDerivedAddresses]
-  );
-
-  const saveEvmSignature = useCallback(
-    (encryptedSignature: string) => {
-      evmDerivedAddresses[evmAddress!].encryptedSignature = encryptedSignature;
-      saveEvmDerivedAddresses(evmDerivedAddresses);
-    },
-    [evmDerivedAddresses, evmAddress]
-  );
-
-  const forgetEvmSignature = useCallback(
-    (_evmAddress = evmAddress) => {
-      if (_evmAddress) {
-        delete evmDerivedAddresses[_evmAddress]?.encryptedSignature;
-        saveEvmDerivedAddresses(evmDerivedAddresses);
-      }
-    },
-    [evmAddress, evmDerivedAddresses]
-  );
-
-  // SOL → dYdX account derivation
-  const [solDerivedAddresses, saveSolDerivedAddresses] = useLocalStorage({
-    key: LocalStorageKey.SolDerivedAddresses,
-    defaultValue: {} as SolDerivedAddresses,
-  });
-
-  const saveSolDerivedAccount = useCallback(
-    ({
-      solAddressInner,
-      dydxAddress,
-    }: {
-      solAddressInner: SolAddress;
-      dydxAddress?: DydxAddress;
-    }) => {
-      saveSolDerivedAddresses({
-        ...solDerivedAddresses,
-        version: LOCAL_STORAGE_VERSIONS[LocalStorageKey.SolDerivedAddresses],
-        [solAddressInner]: {
-          ...solDerivedAddresses[solAddressInner],
-          dydxAddress,
-        },
-      });
-    },
-    [saveSolDerivedAddresses, solDerivedAddresses]
-  );
-
-  const saveSolSignature = useCallback(
-    (encryptedSignature: string) => {
-      solDerivedAddresses[solAddress!].encryptedSignature = encryptedSignature;
-      saveSolDerivedAddresses(solDerivedAddresses);
-    },
-    [solDerivedAddresses, solAddress, saveSolDerivedAddresses]
-  );
-
-  const forgetSolSignature = useCallback(
-    (_solAddress = solAddress) => {
-      if (_solAddress) {
-        delete solDerivedAddresses[_solAddress]?.encryptedSignature;
-        saveSolDerivedAddresses(solDerivedAddresses);
-      }
-    },
-    [solAddress, solDerivedAddresses, saveSolDerivedAddresses]
-  );
-
-  useEffect(() => {
-    // SOL Wallet accounts switched
-    if (previousSolAddress && solAddress && solAddress !== previousSolAddress) {
+    const { address, chain } = sourceAccount;
+    // wallet accounts switched
+    if (previousAddress && address !== previousAddress) {
```

Comment:
is this check needed since we're using `sourceAccount.address` as a dependency for the useEffect?

Reply:
this ones not ideal 😬 so this useEffect is kind of convoluted - it handles both abacus setting and also handling the case of the wallet address changing from the wallet side (and not through disconnecting/connecting through the dapp).

i think as a followup PR, i would like to change the pattern of deriving the localDyDxWallet always from the current sourceAccount.address instead of storing the wallet as state and knowing to set and clear properly